packages/patterns/src/types.js

```diff
@@ -546,8 +580,12 @@ export {};
  *   foo: M.call(AShape, BShape).optional(CShape).rest(EShape).returns(FShape),
  * }
  * ```
- * @property {(rArgGuard: Pattern) => MethodGuardMaker2} rest
- * @property {(returnGuard?: Pattern) => MethodGuard} returns
+ * @property {(restArgGuard: SyncValueGuard) => MethodGuardMaker2} rest
+ * If the rest argument guard is not `M.raw()`, all rest arguments are
+ * automatically hardened and must be Passable.
+ * @property {(returnGuard?: SyncValueGuard) => MethodGuard} returns
+ * If the return guard is not `M.raw()`, the return value is automatically
+ * hardened and must be Passable.
```

Comment:
Is it possible to define these types in a way that doesn't require so much repetition?
```js
/**
 * @typedef {MethodGuardOptMaker & MethodGuardRestMaker & MethodGuardFinalizer} MethodGuardMaker
 * A method name and parameter/return signature like:
 * ```js
 *   foo(a, b, c = d, ...e) => f
 * ```
 * should be guarded by something like:
 * ```js
 * {
 *   ...otherMethodGuards,
 *   foo: M.call(AShape, BShape).optional(CShape).rest(EShape).returns(FShape),
 * }
 * ```
 */

/**
 * @typedef {object} MethodGuardOptMaker
 * @property {(...optArgGuards: ArgGuard[]) => (MethodGuardRestMaker & MethodGuardFinalizer)} optional
 * Optional arguments not guarded with `M.raw()` are automatically hardened and
 * must be Passable.
 */

/**
 * @typedef {object} MethodGuardRestMaker
 * @property {(restArgGuard: SyncValueGuard) => MethodGuardFinalizer} rest
 * If the rest argument guard is not `M.raw()`, all rest arguments are
 * automatically hardened and must be Passable.
 */

/**
 * @typedef {object} MethodGuardFinalizer
 * @property {(returnGuard?: SyncValueGuard) => MethodGuard} returns
 * If the return guard is not `M.raw()`, the return value is automatically
 * hardened and must be Passable.
 */
```